django/contrib/auth/middleware.py

```diff
@@ -34,6 +38,57 @@ def process_request(self, request):
         request.auser = partial(auser, request)
 
 
+class LoginRequiredMiddleware(MiddlewareMixin):
+    """
+    Middleware that forces all views to require login by default.
+
+    Views that have login_not_required decorator will allow unauthenticated
+    requests. Otherwise, it will redirect to a login page.
+    """
+
+    redirect_field_name = REDIRECT_FIELD_NAME
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        if request.user.is_authenticated:
+            return None
+
+        if not getattr(view_func, "login_required", True):
+            return None
+
+        return self.handle_no_permission(request, view_func)
+
+    def get_login_url(self, view_func):
+        login_url = getattr(view_func, "login_url", None) or settings.LOGIN_URL
```

Comment:
Then I'd have the same suggestion here:

```suggestion
        login_url = getattr(view_func, "login_url", settings.LOGIN_URL) 
```

Reply:
(sorry about the previous comment) I tried to update this at the same time, however the `login_url` has a default on `None` when the decorator is used which means this update returns `None` rather than `settings.LOGIN_URL`